drivers: entropy: stm32: extend support to L4 and prevent compilation when not supported

erwango opened the pull request:
This PR requires prior merge of #5420

Enable RNG support on L4 boards by activating 48M clock on which it depends.
Tested successfully on 3 (out of 5 available in Zephyr) STM32L4 based devices.

Enable default support on devices supporting RNG IP.
Prevent compilation on devices not supporting the IP.

Fixes: #5448

erwango committed:
drivers: entropy: stm32: limit driver to SoC supporting it

In STM32 family, RNG IP is available only on F4 and L4 series.
Besides STM32F401 does not support is neither.
Get entropy driver available on STM32 devices supporting it and
generate a compilation issue on STM32 devices that do not own
the RNG IP.

Solves #5448

Signed-Off-By: Erwan Gouriou <erwan.gouriou@linaro.org>

lpereira commented on drivers/entropy/entropy_stm32.c:
```diff
@@ -152,6 +158,30 @@ static int entropy_stm32_rng_init(struct device *dev)
+	LL_RCC_PLLSAI1_EnableDomain_48M();
+
+	/* Wait for PLLSA1 ready flag */
+	while(LL_RCC_PLLSAI1_IsReady() != 1)
```
I can't comment on the STM32 specifics, but there are stylistic issues with this `while` statement here: there must be a space between the keyword and the opening parenthesis; the opening brace should be in the same line as the `while`, and there shouldn't be a semicolon after the closing bracket.

What's the usual time it takes for the PLLSA1 flag to be set? This could impact early boot time since device initialization happens serially.

erwango replied:
@lpereira: Fixing styling issue. Thanks!
About PLLSAI flag, PLLSAI Lock time is max 40µs, typical 15µs. Is that blocking?
If we want RNG, this anyway required. Initialisation level is PRE_KERNEL_2, not sure if it would make sense to change it.

lpereira replied:
40 microseconds boot time increase in the worst case scenario is a good price to pay for a HWPRNG.

erwango committed:
drivers: entropy: Clock activation is missing on stm32l4

On STM32L4 SoCs RNG is clocked by 48MHz domain.
Hence, besides clock activation, it requires 48M
domain to be enabled.

Tested on:
*nucleo_l476rg
*stm32l476g_disco
*disco_l475_iot1

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

pfalcon commented:
@nashif , @galak : Please merge.
